client/wildcard/src/components/NavBar/NavBar.tsx

```diff
@@ -32,6 +32,26 @@ interface NavLinkProps extends NavItemProps, Pick<LinkProps<H.LocationState>, 't
     external?: boolean
 }
 
+const useOutsideClickDetector = (
+    reference: React.RefObject<HTMLUListElement>
+): [boolean, React.Dispatch<React.SetStateAction<boolean>>] => {
+    const [outsideClick, setOutsideClick] = useState(() => true)
+
+    useEffect(() => {
+        function handleClickOutside(event: MouseEvent): void {
+            if (reference.current && !reference.current.contains(event.target as Node | null)) {
+                setOutsideClick(() => false)
```

Comment:
```suggestion
    const [outsideClick, setOutsideClick] = useState(true)

    useEffect(() => {
        function handleClickOutside(event: MouseEvent): void {
            if (reference.current && !reference.current.contains(event.target as Node | null)) {
                setOutsideClick(false)
```

Don't think we need a func here?